### DISCUSSION TITLE
Guidance on upgrading legacy project plus ideas on potential bug

### DISCUSSION THREAD

**Discussion by developedbyprince** [mozilla/pdf.js#18582] [General]
We have a production level solution using an OLD version of PDFjs, 1.6.52 (yeah I know, ROUGH). For the first time in years, I've encountered an issue where parts of a document isn't being rendered properly, mainly where the font changes. Here's an obscured screenshot of the actual pdf (right) vs rendered left. 
Notice the names on the right do not render on the left but the labels do, I assume do to some font issue? Any ideas what could cause certain sections of a PDF not to render.
![image](https://github.com/user-attachments/assets/deaf68dc-a800-4447-b920-95f8dce6a528)

Secondly, It makes sense to update the library however that would require a lot of effort and I'm not sure if the functionality we need can be achieved. Right now, we use version 1.6.52 to load multiple PDFs on the same page, that the end user can scroll through seamlessly, as they reach the end of the first document, the next page is the start of the next document. I see currently in the demo's it's like a single file only at a time sort of rendering. Is what I need achievable and any insight to achieving such (rendering hooks?)?

I do intend to upgrade but due to time if any insight can be provided for the potential "font" issue that'll be great. Thanks.

**Comment by calixteman** [comment]
You can try to get pdf.js locally and try to git bisect to find the patch which fixed your issue and see if it's easy to backport.
